### DISCUSSION TITLE
Custom extension using directus native components

### DISCUSSION THREAD

**Discussion by bastibuck** [directus/directus#13434] [Archive]
Hey,

so I have just implemented my first custom extension (a custom markdown interface based on the original markdown interface). I'm using a bunch of directus "native" components e.g. `<v-dialog />`, `<v-input />`etc. since they integrate very nicely with the look and feel of the whole app. 

Is this a bad idea? Regarding future updates possibly breaking my extension

Couldn't find any dedicated Changelog messages about changes in these components nor recommendations to use these in a custom extension. 

Thanks

**Comment by rijkvanzanten** [comment]
I don't think it's a bad idea! We made them globally registered for this exact use case. The base components (the ones starting with `v-`) don't really change all that often. That being said, depending on the usage of those components, you should ensure that there weren't any unexpected changes that affect how you're using these exact components between versions, as we don't treat any changes in those components as breaking changes. (For example, changing the default color for v-button from `var(--foreground)` to `currentColor` won't be considered a noteworthy change, but might break things in your extension depending on the exact usage of the component).

**Reply by bastibuck** [reply]
Will have to scan the Changelog than very carefully. Thank you
